src/machines/settingsMachine.ts

```diff
@@ -9,177 +9,187 @@ import {
-    id: 'Settings',
-    predictableActionArguments: true,
+export const settingsMachine = setup({
+  types: {
     context: {} as ReturnType<typeof createSettings>,
-    initial: 'idle',
-    states: {
-      idle: {
-        entry: ['setThemeClass', 'setClientSideSceneUnits'],
-
-        on: {
-          '*': {
-            target: 'persisting settings',
-            actions: ['setSettingAtLevel', 'toastSuccess'],
-          },
-
-          'set.app.onboardingStatus': {
-            target: 'persisting settings',
-
-            // No toast
-            actions: ['setSettingAtLevel'],
-          },
-
-          'set.app.themeColor': {
-            target: 'persisting settings',
-
-            // No toast
-            actions: ['setSettingAtLevel'],
-          },
-
-          'set.modeling.defaultUnit': {
-            target: 'persisting settings',
-
-            actions: [
-              'setSettingAtLevel',
-              'toastSuccess',
-              'setClientSideSceneUnits',
-              'Execute AST',
-            ],
-          },
-
-          'set.app.theme': {
-            target: 'persisting settings',
-
-            actions: [
-              'setSettingAtLevel',
-              'toastSuccess',
-              'setThemeClass',
-              'setEngineTheme',
-              'setClientTheme',
-            ],
-          },
-
-          'set.app.streamIdleMode': {
-            target: 'persisting settings',
-
-            actions: ['setSettingAtLevel', 'toastSuccess'],
-          },
-
-          'set.modeling.highlightEdges': {
-            target: 'persisting settings',
-
-            actions: ['setSettingAtLevel', 'toastSuccess', 'setEngineEdges'],
-          },
-
-          'Reset settings': {
-            target: 'persisting settings',
-
-            actions: [
-              'resetSettings',
-              'setThemeClass',
-              'setEngineTheme',
-              'setClientSideSceneUnits',
-              'Execute AST',
-              'setClientTheme',
-            ],
-          },
-
-          'Set all settings': {
-            actions: [
-              'setAllSettings',
-              'setThemeClass',
-              'setEngineTheme',
-              'setClientSideSceneUnits',
-              'Execute AST',
-              'setClientTheme',
-            ],
-          },
-
-          'set.modeling.showScaleGrid': {
-            target: 'persisting settings',
-            actions: [
-              'setSettingAtLevel',
-              'toastSuccess',
-              'setEngineScaleGridVisibility',
-            ],
-          },
+    input: {} as ReturnType<typeof createSettings>,
```

Comment:
I'm still not entirely understanding this new layer of `input`s, might be something I can talk to you about later after I've given the docs another read.

Reply:
Happy to chat!